dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ResolveNonDetChoices.java

```diff
@@ -0,0 +1,46 @@
+package com.dat3m.dartagnan.program.processing;
+
+import com.dat3m.dartagnan.expression.Expression;
+import com.dat3m.dartagnan.expression.Type;
+import com.dat3m.dartagnan.expression.processing.ExpressionInspector;
+import com.dat3m.dartagnan.program.Function;
+import com.dat3m.dartagnan.program.Program;
+import com.dat3m.dartagnan.program.event.EventFactory;
+import com.dat3m.dartagnan.program.event.core.Local;
+import com.dat3m.dartagnan.program.event.lang.svcomp.NonDetChoice;
+import com.dat3m.dartagnan.program.misc.NonDetValue;
+
+public class ResolveNonDetChoices implements ProgramProcessor {
```

Comment:
Why is this a program processor if we just propagate the call to each function?

Reply:
Because on hand it only runs on threads (it is almost always wrong to run it on functions) and one the other hand it affects the program by generating new program constants.
In particular, the pass could never run on an isolated function that does not belong to some program.